m4opt/models/extinction/airmass.py

```diff
@@ -0,0 +1,394 @@
+    >>> with state.set_observing(target_coord=target, obstime=time):
+    ...     print(extn(3200*u.Angstrom))
+    0.23643960524293295
+
+    `from_observer()` also has the capability to set the airmass model:
+
+    >>> from m4opt.models.extinction import KastenYoungAirmass
+    >>> extn = AtmosphericExtinction.from_observer(place, \
+        airmass_class=KastenYoungAirmass)
+    >>> with state.set_observing(target_coord=target, obstime=time):
+    ...     print(extn(3200*u.Angstrom))
+    0.2369337657824151
+
+    If we already have an airmass object instantiated,
+    we can initialize with it instead:
+
+    >>> extn = AtmosphericExtinction.from_airmass(airmass)
+    >>> with state.set_observing(target_coord=target, obstime=time):
+    ...     print(extn(3200*u.Angstrom))
+    0.23643960524293295
+
+    Or we can initialize a blank state, and fill in the blanks later:
+
+    >>> extn = AtmosphericExtinction()
+    >>> with state.set_observing(target_coord=target, obstime=time, \
+                                 observatory_loc=place):
+    ...     print(extn(3200*u.Angstrom))
+    0.23643960524293295
+
+    You can also define the airmass model for this blank state:
+
+    >>> extn = AtmosphericExtinction(airmass_class=KastenYoungAirmass)
+    >>> with state.set_observing(target_coord=target, obstime=time, \
+                                 observatory_loc=place):
+    ...     print(extn(3200*u.Angstrom))
+    0.2369337657824151
+
+    All AtmosphericExtinction models require an extinction table. The
+    models above have used the default table; however, other tables
+    can be chosen at initialization by passing the appropriate parameter:
+
+    >>> AtmosphericExtinction.available_tables
+    {'apo': 'apoextinct.dat',\
+ 'kpno': 'kpnoextinct.dat',\
+ 'ctio': 'ctioextinct.dat',\
+ 'lapalm': 'lapalmextinct.dat',\
+ 'mko': 'mkoextinct.dat',\
+ 'lick': 'mthamextinct.dat',\
+ 'paranal': 'paranalextinct.dat'}
+
+    >>> extn = AtmosphericExtinction.at(airmass, target, time, \
+        table_name='kpno')
+    >>> extn(3200*u.Angstrom)
+    <Quantity 0.23643961>
+
+    >>> extn = AtmosphericExtinction(table_name='apo')
+    >>> with state.set_observing(target_coord=target, obstime=time, \
+                                 observatory_loc=place):
+    ...     print(extn(3200*u.Angstrom))
+    0.2651060248681333
+
+    Finally, we can pass in the interpolation method for the extinction table
+    if finer control is needed. The default is 'linear', but other options can
+    be selected (see `astropy.modeling.tabular.Tabular1D` for details).
+
+    >>> extn = AtmosphericExtinction(table_name='apo', table_method='nearest')
+    >>> with state.set_observing(target_coord=target, obstime=time, \
+                                 observatory_loc=place):
+    ...     print(extn(3200*u.Angstrom))
+    0.2727180718446889
+    """
+
+    def __new__(cls, table_name=None, table_method="linear",
+                airmass_class=Airmass):
```

Comment:
Instead of an airmass class, let's just take an airmass instance --- which could be simply a quantity, for known airmass.

Reply:
I will go ahead and make the default `__new__()` initializer require an airmass instance rather than being a "blank" initializer. 

However, I will keep the functions that require a `Airmass` class instance, since these are required if we will have `observatory_loc` as part of the `ScienceState`. I suggest revisiting the `ScienceState` question at a later time.